ocaml/typing/typecore.ml

```diff
@@ -475,7 +475,7 @@ let type_constant = function
 
 let type_constant_unboxed env loc
     : Jane_syntax.Unboxed_constants.expression -> _ = function
-  | Float _ -> instance Predef.type_float_unboxed
+  | Float _ -> instance Predef.type_unboxed_float
```

Comment:
As discussed in the meeting, this looks like it's problematic, because we'll have values of type `float#` that aren't actually treated as unboxed. Maybe just change this to be `Predef.type_float` for now.

Reply:
This turned out not to be reachable, because `unboxed_constant_or_raised` is called first and it rejects unboxed float literals.  So, no bug, but I've changed this to also explicitly raise just for clarity.